operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-federation-frontend-values.yaml

```diff
@@ -0,0 +1,68 @@
+# Enable enterprise features (required for federation-frontend)
+enterprise:
+  enabled: true
+
+# Enable and configure federation-frontend
+federation_frontend:
+  enabled: true
+  replicas: 2
+  resources:
+    requests:
+      cpu: 100m
+      memory: 128Mi
+    limits:
+      cpu: 1
+      memory: 256Mi
+
+mimir:
+  structuredConfig:
+    auth:
+      type: trust
+    instrumentation:
+      enabled: false
+    federation:
+      proxy_targets:
+        - name: "mimir-distributed-1"
+          url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
+        - name: "mimir-distributed-2"
+          url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
```

Comment:
This isn't necessarily something for this PR, but have you considered making this configurable as a value underneath the `federation_frontend` value? Given it will always need to be set when the federation-frontend is deployed, might make things a little simpler.

Reply:
Soo:


```yaml
federation_frontend:
  proxy_targets:
    - name: "mimir-distributed-1"
      url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
    - name: "mimir-distributed-2"
      url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
```

vs

```yaml
mimir:
  structuredConfig:
      federation:
        proxy_targets:
          - name: "mimir-distributed-1"
            url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
          - name: "mimir-distributed-2"
            url: "http://mimir-query-frontend.real-deployment.svc.cluster.local:8080/prometheus"
```


I spent about 20 min trying to get the validation to work ok so users can't provide both (because one is silently dropped then). It's surprisingly difficult and it's still not robust enough.

Do you think it will make a big enough difference? I'm erring on the side of not doing it.

Reply:
> although I imagine the same validation problem occurs with all the other options we configure automatically.

the fact that this is a list makes it more complicated, the fact that there's a default empty list value. Otherwise we overlap the two configs if they're only nested objects without lists.

Reply:
I think this is fine to leave as you have it.